includes/wikia/parser/templatetypes/RecognizedTemplatesProvider.class.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/**
+ * Class provides methods to retrieve a list of templates defined as recognized
+ * Also provides possibility to restrict list of templates to those used only on pages within certain namespaces
+ * (both directly and indirectly)
+ * e.g. wgContentNamespaces
+ */
+
+class RecognizedTemplatesProvider {
+	const ERROR_MESSAGE = 'ExternalTemplatesProviderError';
+
+	private $tcs;
+	private $wikiaId;
+	private $namespaces;
+	private $namespacesTemplates;
+
+	public function __construct( TemplateClassificationService $tcs, $wikiaId, $namespaces = [] ) {
```

Comment:
I think that we should let the marketing language creep into engineering choices and make it `$wikiId` :)